src/components/Modal.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from "react";
+import Backdrop from "@mui/material/Backdrop";
+import Box from "@mui/material/Box";
+import Modal from "@mui/material/Modal";
+import Button from "@mui/material/Button";
+import { useSpring, animated } from "@react-spring/web";
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import IconButton from '@mui/material/IconButton';
+import CloseIcon from '@mui/icons-material/Close';
+
+interface FadeProps {
```

Comment:
Your interfaces should be defined in their own files, your code will be cleaner this way